apps/wallet-ui/src/components/ConnectButton.tsx

```diff
@@ -0,0 +1,41 @@
+"use client";
+import {
+  generatePayload,
+  getCurrentUser,
+  login,
+  logout,
+} from "@/app/[ecosystem]/auth";
+import { client } from "@/lib/client";
+import { useTheme } from "next-themes";
+import type { VerifyLoginPayloadParams } from "thirdweb/auth";
+import {
+  arbitrum,
+  base,
+  blast,
+  mainnet,
+  optimism,
+  zkSync,
+} from "thirdweb/chains";
+import { ConnectButton as ThirdwebConnectButton } from "thirdweb/react";
+import { ecosystemWallet } from "thirdweb/wallets";
+
+export default function ConnectButton({
+  ecosystem,
+}: { ecosystem: `ecosystem.${string}` }) {
+  const { theme } = useTheme();
+
+  return (
+    <ThirdwebConnectButton
+      chains={[mainnet, base, optimism, arbitrum, blast, zkSync]}
+      wallets={[ecosystemWallet(ecosystem)]}
+      client={client}
+      theme={theme as "light" | "dark"}
+      auth={{
```

Comment:
once again questioning if we truly _need_ this here

Reply:
Yes because the site should never allow solely a wallet connection without auth

Reply:
yes but to enforce that you already check headers on every page?